src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,200 @@
+                if (itemNode == first) {
+                    first = first.next;
+                    if (first != null) {
+                        first.prev = null;
+                    } else {
+                        last = null;
+                    }
+                } else if (itemNode == last) {
+                    last = last.prev;
+                    if (last != null) {
+                        last.next = null;
+                    } else {
+                        first = null;
+                    }
+                } else {
+                    unlink(itemNode);
+                }
+                size--;
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (first == null && last == null) {
+            return true;
+        }
+        return size == 0;
+    }
+
+    private void linkLast(T element) {
+        final Node<T> prev = last;
+        final Node<T> newNode = new Node<>(prev,element,null);
+        last = newNode;
+        if (prev == null) {
+            first = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        size++;
+    }
+
+    private void checkPositionIndex(int index, int expressionToCheck) {
+        if (expressionToCheck == NOT_SETTING) {
+            if (!isPositionIndex(index)) {
+                throw new IndexOutOfBoundsException(createIndexOutOfBoundMessage(index));
+            }
+        } else {
+            if (!isPositionIndexToSet(index)) {
+                throw new IndexOutOfBoundsException(createIndexOutOfBoundMessage(index));
+            }
+        }
+    }
+
+    private Node<T> node(int index) {
+        if (index < (size >> 1)) {
+            Node<T> current = first;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            return current;
+        } else {
+            Node<T> current = last;
+            for (int i = size - 1; i > index; i--) {
+                current = current.prev;
+            }
+            return current;
+        }
     }
+
+    private void linkBefore(T value, Node<T> node) {
+        if (node != null) {
+            final Node<T> pred = node.prev;
+            final Node<T> newNode = new Node<>(pred, value, node);
+            node.prev = newNode;
+            if (pred == null) {
+                first = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            size++;
+        }
+    }
+
+    private void unlink(Node<T> node) {
+        Node<T> pred = node.prev;
+        Node<T> next = node.next;
+        pred.next = next;
+        next.prev = pred;
+    }
+
+    private String createIndexOutOfBoundMessage(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
```

Comment:
It is not necessary to have method to create exceptions message, even they have same logic, but it is rule only for exceptions.